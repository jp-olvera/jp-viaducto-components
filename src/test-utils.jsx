import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
