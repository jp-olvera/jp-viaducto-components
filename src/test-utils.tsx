import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
