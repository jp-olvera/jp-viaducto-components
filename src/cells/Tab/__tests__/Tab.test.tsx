/* eslint-env jest */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/react';
import { render, screen, fireEvent } from '../../../test-utils';
import { Tab } from '..';

describe('<Tab/>', () => {
  const handleFn = jest.fn();

  test('should render properly', () => {
    render(
      <Tab
        text='Tab'
        type='warning'
        horizontalSpacing='sm'
        verticalSpacing='aaaa'
        icon='❤'
        active
      />,
    );
    expect(screen.queryByText('Tab')).toBeVisible();
  });

  test('should call function', () => {
    const { getByText } = render(
      <Tab
        text='tab'
        onClick={handleFn}
        icon='❤'
        lead
        verticalSpacing={null}
        horizontalSpacing={null}
      />,
    );
    fireEvent.click(getByText('tab'));
    expect(handleFn).toHaveBeenCalled();
  });

  test('should render default props', () => {
    const { container } = render(<Tab />);
    expect(container).toMatchSnapshot();
  });

  test('should render with top line', () => {
    const { container } = render(<Tab text='top' position='top' />);
    expect(container).toMatchSnapshot();
  });

  test('should render with left line', () => {
    const { container } = render(<Tab text='left' position='left' />);
    expect(container).toMatchSnapshot();
  });

  test('should render with right line', () => {
    const { container } = render(<Tab text='right' position='right' />);
    expect(container).toMatchSnapshot();
  });
});
