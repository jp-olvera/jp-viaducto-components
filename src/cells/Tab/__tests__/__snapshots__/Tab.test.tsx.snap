// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab/> should render default props 1`] = `
<div>
  <button
    class="StyledTab-sc-7ood0w-0 eEDjph"
    color="#F1F1F1"
  >
    <div
      class="tab-text"
      style="display: flex; align-items: center;"
    >
      <p
        class="StyledParagraph-sc-dbd4lw-0 gKKAmW"
        color="darkGray"
      >
        Tab
      </p>
    </div>
  </button>
</div>
`;

exports[`<Tab/> should render with left line 1`] = `
<div>
  <button
    class="StyledTab-sc-7ood0w-0 LjWER"
    color="#F1F1F1"
  >
    <div
      class="tab-text"
      style="display: flex; align-items: center;"
    >
      <p
        class="StyledParagraph-sc-dbd4lw-0 gKKAmW"
        color="darkGray"
      >
        left
      </p>
    </div>
  </button>
</div>
`;

exports[`<Tab/> should render with right line 1`] = `
<div>
  <button
    class="StyledTab-sc-7ood0w-0 ivsheN"
    color="#F1F1F1"
  >
    <div
      class="tab-text"
      style="display: flex; align-items: center;"
    >
      <p
        class="StyledParagraph-sc-dbd4lw-0 gKKAmW"
        color="darkGray"
      >
        right
      </p>
    </div>
  </button>
</div>
`;

exports[`<Tab/> should render with top line 1`] = `
<div>
  <button
    class="StyledTab-sc-7ood0w-0 hqlCyh"
    color="#F1F1F1"
  >
    <div
      class="tab-text"
      style="display: flex; align-items: center;"
    >
      <p
        class="StyledParagraph-sc-dbd4lw-0 gKKAmW"
        color="darkGray"
      >
        top
      </p>
    </div>
  </button>
</div>
`;
